lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -85,55 +134,14 @@ private void Initialize()
         /// <returns></returns>
         public T GetPartitionHandle()
         {
-            if (_copyToLocal)
-            {
-                if (!_isInitialized)
-                {
-                    Initialize();
-                }
-
-                return _fileSerializer.Deserialize(_localFiles);
-            }
-            return _fileSerializer.Deserialize(_filePaths);
-        }
-
-        private void CopyFromRemote()
-        {
-            string localFileFolder = _tempFileCreator.CreateTempDirectory("-partition-");
-            Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "Local file temp folder: {0}", localFileFolder));
-
-            foreach (var sourceFilePath in _filePaths)
+            lock (_lock)
             {
-                Uri sourceUri = _fileSystem.CreateUriForPath(sourceFilePath);
-                Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "sourceUri {0}: ", sourceUri));
-                if (!_fileSystem.Exists(sourceUri))
+                if (!_localFiles.IsPresent())
```

Comment:
Can you clarify the definition of "cache"? Did you provide any guarantees for your users? It seems that in the previous iteration, the user is not guaranteed anything about the location of the data.